File: VB/CustomWizardExample/Form1.vb
Imports DevExpress.Snap.Services
Imports CustomWizardExample.Wizard
Imports DevExpress.Office.Options

Namespace CustomWizardExample
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()
            InitializeSnapControl()
        End Sub
        Private Sub InitializeSnapControl()
            snapControl.Text = "Click the ""Add New Data Source"" button to invoke the customized Data Source Wizard that allows you to establish a connection to an XML data file."
            snapControl.Options.DataSourceWizardOptions.DataSourceTypes = DataSourceTypes.Sql Or DataSourceTypes.Excel
            snapControl.AddService(GetType(IDataSourceWizardCustomizationService), New DataSourceWizardCustomizationService())
            snapControl.AddService(GetType(IDataSourceCreationService), New CustomDataSourceCreationService())
        End Sub
    End Class
End Namespace

File: CS/CustomWizardExample/Form1.cs
using DevExpress.Snap.Services;
using CustomWizardExample.Wizard;
using DevExpress.Office.Options;

namespace CustomWizardExample
{
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm
    {
        public Form1() {
            InitializeComponent();
            InitializeSnapControl();
        }
        void InitializeSnapControl() {
            snapControl.Text = "Click the \"Add New Data Source\" button to invoke the customized Data Source Wizard that allows you to establish a connection to an XML data file.";
            snapControl.Options.DataSourceWizardOptions.DataSourceTypes = DataSourceTypes.Sql | DataSourceTypes.Excel;
            snapControl.AddService(typeof(IDataSourceWizardCustomizationService), new DataSourceWizardCustomizationService());
            snapControl.AddService(typeof(IDataSourceCreationService), new CustomDataSourceCreationService());
        }
    }
}
